apps/server/src/database/queries/stats.ts

```diff
@@ -13,14 +12,32 @@ import {
   sortByTimeSplit,
 } from "./statsTools";
 
+export type ItemType = {
+  field: string;
+};
+
+export const itemTypes: {
+  [key in "track" | "album" | "artist"]: ItemType;
+} = {
+  track: {
+    field: "$id",
+  },
+  album: {
+    field: "$albumId",
+  },
+  artist: {
+    field: "$primaryArtistId",
+  },
+};
```

Comment:
I think there is a more modern way to define this.
They are constants so they deserve a capital and don't need to be plural, I feel like **ItemType** is better and it should not collide with the ItemType type since it's a value. Also, you don't need to set the keys in the type declaration, you can just do
```typescript

export const ItemType = {
  track: {
    field: "$id",
  },
  album: {
    field: "$albumId",
  },
  artist: {
    field: "$primaryArtistId",
  },
} as const satisfies Record<string, ItemType>;
```
so you can later do ItemType.artist, while ItemType.heyhey won't work.